daemon/src/maker_cfd_actor.rs

```diff
@@ -169,11 +169,19 @@ impl MakerCfdActor {
 
         let (actor, inbox) = setup_contract_actor::new(
             {
-                let inbox = self.takers()?.clone();
+                let takers = self.takers()?.clone();
                 move |msg| {
-                    inbox.send(maker_inc_connections_actor::TakerMessage {
-                        taker_id,
-                        command: TakerCommand::OutProtocolMsg { setup_msg: msg },
+                    tokio::spawn({
+                        let inbox = takers.clone();
+                        async move {
+                            inbox
+                                .do_send_async(maker_inc_connections_actor::TakerMessage {
+                                    taker_id,
+                                    command: TakerCommand::OutProtocolMsg { setup_msg: msg },
+                                })
+                                .await
+                                .unwrap();
+                        }
                     });
```

Comment:
There is no need to spawn this into the runtime, sending the message doesn't take long so you can easily to this in the parent async context. Just append an `.await` to the `send` function.

Reply:
hm, this is a closure, so there's no parent async context unfortunately :(